core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java

```diff
@@ -31,18 +30,19 @@ public static Envelope from(
             TrainPath trainPath,
             RollingStock rollingStock,
             boolean addRollingStockLength,
-            String tag
-    ) {
-        return from(TrainPath.removeLocation(trainPath.trackRangePath()), rollingStock, addRollingStockLength, tag);
+            String tag,
+            DriverBehaviour driverBehaviour) {
+        return from(TrainPath.removeLocation(trainPath.trackRangePath()), rollingStock, addRollingStockLength, tag,
+                driverBehaviour);
```

Comment:
I would be more comfortable if we keep this out of the MRSP, it would be better as an extra step between MRSP and max effort envelope. Unless we want this to be part of the MRSP displayed in the front-end? (@axrolld)

The way I see it, this could be the responsibility of the DriverBehavior object. We could imagine something like 
`var MRSPWithDriverBehavior = driverBehavior.applyToMRSP(mrsp);` as an extra step. In this case, all the fields should remain private.

Reply:
No, indeed this doesn't need to be displayed in the speed limits in front end. The idea here was to implement the offset the same way as the one corresponding tot the train length, but i agree with your proposal of adding an extra step, that seems cleaner.

Reply:
Although, the acceleration offset due to the train length have been implemented this way, however it's not visible in the speed limits in the front-end, so maybe if it's implemented this way the driver behavior's offsets also won't be part of the speed limits in the front end ?

Reply:
Then why can't we just keep those two MRSP, one displayed in the front end, and the other with the train length + driver behavior offset ?

Reply:
We can, it's about splitting this into two different steps. It makes it much easier to write tests, and it makes driver behavior more "contained". If we patch every step of the pipeline with extra responsibilities for every "driver behavior" it will soon become an unmaintainable mess